ocs_ci/ocs/scale_noobaa_lib.py

```diff
@@ -67,28 +67,33 @@ def check_all_obc_reached_bound_state_in_kube_job(
     """
     # Check all OBCs to reach Bound state
     obc_bound_list, obc_not_bound_list = ([] for i in range(2))
-    while_iteration_count = 0
+    while_iteration_count_1 = 0
+    while_iteration_count_2 = 0
     while True:
-        # Get kube_job obj and fetch either all OBC's are in Bound state
+        # Get kube_job obj and check if all OBC's are in Bound state
         # If not bound adding those OBCs to obc_not_bound_list
-        job_get_output = kube_job_obj.get(namespace=namespace)
-        for i in range(no_of_obc):
-            status = job_get_output["items"][i]["status"]["phase"]
-            log.info(
-                f"obc {job_get_output['items'][i]['metadata']['name']} status {status}"
-            )
-            if status != constants.STATUS_BOUND:
-                obc_not_bound_list.append(
-                    job_get_output["items"][i]["metadata"]["name"]
-                )
+        job_get_output = kube_job_obj.get(namespace=namespace).get("items")
+        if job_get_output is not None and len(job_get_output) == no_of_obc:
+            for i in range(no_of_obc):
+                status = job_get_output[i]["status"]["phase"]
+                log.info(f"obc {job_get_output[i]['metadata']['name']} status {status}")
+                if status != constants.STATUS_BOUND:
+                    obc_not_bound_list.append(job_get_output[i]["metadata"]["name"])
+                    # Wait 10 secs to ensure the next obc on the list has status field populated
+                    time.sleep(10)
+                    job_get_output = kube_job_obj.get(namespace=namespace).get("items")
+        else:
+            while_iteration_count_1 += 1
+            time.sleep(timeout)
+            continue
```

Comment:
@tiffanyn108 Hope this worked to solve the issue we discussed.

Reply:
Yes, it is.